src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepositoryTest.java

```diff
@@ -91,8 +94,49 @@ public void setUp() {
 
         switchA = createTestSwitch(TEST_SWITCH_A_ID.getId());
         switchB = createTestSwitch(TEST_SWITCH_B_ID.getId());
+        switchC = createTestSwitch(TEST_SWITCH_C_ID.getId());
 
-        assertEquals(2, switchRepository.findAll().size());
+        assertEquals(3, switchRepository.findAll().size());
+    }
+
+    @Test
+    public void findPortToFlowsMap() {
+        createTestFlow(TEST_FLOW_ID, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_2, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_3, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_4, switchC, switchB);
+
+        Map<Integer, List<Flow>> result = flowRepository.findSwitchFlowsByPort(switchA.getSwitchId(), null);
+
+        assertNotNull(result);
+        assertTrue("The map must contain a key for the port", result.containsKey(PORT_1));
+        assertNotNull("The map must contain a non-null value for the test port", result.get(PORT_1));
+        assertEquals("The map must contain exactly two keys", 2, result.size());
+        assertEquals("The map must contain exactly 3 flows", 3, result.get(PORT_1).size());
```

Comment:
"The map must contain exactly 3 flows" this comment is misleading. Here, you are asserting that the value 
 of PORT_1 contains 3 flows, not the map itself"

Reply:
done